### DISCUSSION TITLE
Trying to add interactive markdown tables with DataTables library

### DISCUSSION THREAD

**Discussion by Krajzys** [requarks/wiki#6506] [Help / Questions] [answered]
Hi, I'm trying to add interactive tables using [DataTables](https://datatables.net/manual/installation) library to our wiki.js. I added the necessary code in the administration panel to inject it at the end of the head tag.
In this way when I add the class `interactive-table` to a markdown table, it should become interactible (with options to sort and filter).

My problem is that this doesn't work on page load.
If I do it manually by running this `$('.interactive-table').DataTable()` in the console it starts to work.
But it would be best if those could work immediately after loading the page to the user.

Does anyone have experience with DataTables and could help me out here?

Code injected at the end of head tag
```html
<link rel="stylesheet" href="https://cdn.datatables.net/1.13.4/css/jquery.dataTables.css" />
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.4/jquery.min.js"></script>
<script src="https://cdn.datatables.net/1.13.4/js/jquery.dataTables.js"></script>

<script>
$().ready(() => {
  $('.interactive-table').DataTable()
})
</script>
```

Example page to test out DataTables
```markdown
# Interactive Table

|Name|Surname|Phone No.|
|-|-|-|
|John|Smith|111111111|
|Michael|Angelo|222222222|
|Leonardo|DaVinci|333333333|
{.interactive-table}

```

**Comment by fnafox** [comment]
Try entering this test into the page specific script override (Edit > Page Properties > Scripts)
```html
<script>
$().ready(() => {
  $('.interactive-table').DataTable()
})
</script>
```

**Reply by NGPixel** [reply, marked as answer]
That's not the correct way, you need to use the `page-ready` event:

```
window.boot.register('page-ready', () => {
   $('.interactive-table').DataTable()
})
```

**Reply by Krajzys** [reply]
That helped 👍 
Thanks a lot!
